transport/src/main/java/io/netty/channel/oio/AbstractOioByteChannel.java

```diff
@@ -203,7 +204,15 @@ protected void doWrite(ChannelOutboundBuffer in) throws Exception {
                     readableBytes = newReadableBytes;
                 }
                 in.remove();
+            } else if (msg instanceof AbstractFileRegion) {
+                AbstractFileRegion region = (AbstractFileRegion) msg;
+                long transferIndex = region.transferIndex();
+                doWriteFileRegion(region);
+                in.progress(region.transferIndex() - transferIndex);
+                in.remove();
             } else if (msg instanceof FileRegion) {
+                // TODO: consider remove this branch once we move the methods in AbstractFileRegion
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1135)